src/components/map/MapService.js

```diff
@@ -2473,5 +2515,75 @@ goog.require('ga_urlutils_service');
       return new PreviewLayers();
     };
   });
+
+  /**
+   * Service to pseudo-hide layers behind fully opaque layers
+   * to avoid loading tiles/data for such layers.
+   */
+  module.provider('gaLayerHideManager', function() {
+    this.$get = function($rootScope, $timeout, gaLayers, gaLayerFilters) {
+      var layers = [];
+      var unregKeys = [];
+
+      var unreg = function() {
+        angular.forEach(unregKeys, function(key) {
+          if (key) {
+            ol.Observable.unByKey(key);
+          }
+        });
+        unregKeys = [];
+      };
+
+      var updateHiddenState = function() {
+        var sortedLayers = layers.slice().reverse();
+        var hide = false;
+
+        unreg();
+
+        angular.forEach(sortedLayers, function(layer) {
+          layer.hiddenByOther = hide;
+
+          // Register all layers for changes
+          unregKeys.push(layer.on('change:visible', function(evt) {
+            $timeout(function() {
+              updateHiddenState();
+            }, 0);
+          }));
+          unregKeys.push(layer.on('change:opacity', function(evt) {
+            $timeout(function() {
+              updateHiddenState();
+            }, 0);
+          }));
+
+          // First opaque layer
+          if (!hide &&
+              gaLayers.getLayer(layer.id) &&
+              gaLayers.getLayer(layer.id).opaque &&
+              layer.visible &&
+              layer.invertedOpacity == '0') {
+            hide = true;
+          }
+        });
+      };
+
+      return function(map) {
+        var scope = $rootScope.$new();
+        scope.layers = map.getLayers().getArray();
+        scope.f = function(l) {
+          if (gaLayerFilters.background(l) ||
+              gaLayerFilters.selected(l)) {
+            return true;
+          }
+          return false;
```

Comment:
You can return  directly `return (gaLayerFilters.background(l) || gaLayerFilters.selected(l));`-,